WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -294,9 +293,15 @@ public static void cancelFinalNotificationForMedia(Context context, SiteModel si
         PostUploadNotifier.cancelFinalNotificationForMedia(context, site);
     }
 
+    /**
+     * Do not use this method unless the user explicitly confirmed changes - eg. clicked on publish button or
+     * similar.
+     */
     private void makePostPublishable(@NonNull PostModel post) {
         PostUtils.updatePublishDateIfShouldBePublishedImmediately(post);
         post.setStatus(PostStatus.PUBLISHED.toString());
+        AppLog.d(T.POSTS, "Changes explicitly confirmed by the user. Post Title: " + post.getTitle());
+        post.setChangesConfirmedContentHashcode(post.contentHashcode());
```

Comment:
This method is used when the user clicks on Publish action on a system notification. Actions in system notifications are invoked using a PendingIntent. The PendingIntent starts the UploadService directly therefore we need to update the hashcode (confirm the changes) here.